vertx-pg-client/src/main/java/io/vertx/pgclient/impl/PgConnectionImpl.java

```diff
@@ -34,30 +34,27 @@
 import io.vertx.core.Handler;
 import io.vertx.core.Vertx;
 import io.vertx.pgclient.impl.codec.TxFailedEvent;
-import io.vertx.sqlclient.impl.tracing.QueryTracer;
+
+import java.util.function.Supplier;
 
 public class PgConnectionImpl extends SqlConnectionBase<PgConnectionImpl> implements PgConnection  {
 
-  public static Future<PgConnection> connect(ContextInternal context, PgConnectOptions options) {
-    if (options.isUsingDomainSocket() && !context.owner().isNativeTransportEnabled()) {
-      return context.failedFuture("Native transport is not available");
-    } else {
-      PgConnectionFactory client;
-      try {
-        client = new PgConnectionFactory(context.owner(), options);
-      } catch (Exception e) {
-        return context.failedFuture(e);
-      }
-      context.addCloseHook(client);
-      return (Future) client.connect(context);
+  public static Future<PgConnection> connect(ContextInternal context, Supplier<PgConnectOptions> options) {
```

Comment:
Is there a reason for changing the type of the `options` parameter to `Supplier`?

This method is only used for one-shot connection, and the only caller creates a synthetic supplier

Reply:
it might be a left over of previous architecture, if you can simplify it go ahead